tests/dbus/test_dbus_client.cpp

```diff
@@ -182,6 +187,24 @@ void CheckSrpServerInfo(ThreadApiDBus *aApi)
     TEST_ASSERT(srpServerInfo.mResponseCounters.mOther == 0);
 }
 
+void CheckTrelInfo(ThreadApiDBus *aApi)
+{
+    OTBR_UNUSED_VARIABLE(aApi);
+
+#if OTBR_ENABLE_TREL
+    TrelInfo trelInfo;
+    TEST_ASSERT(aApi->GetTrelInfo(trelInfo) == OTBR_ERROR_NONE);
+
+    TEST_ASSERT(trelInfo.mEnabled);
+    TEST_ASSERT(trelInfo.mNumTrelPeers == 0);
+    TEST_ASSERT(trelInfo.mTrelCounters.mRxBytes == 0);
+    TEST_ASSERT(trelInfo.mTrelCounters.mRxPackets == 0);
+    TEST_ASSERT(trelInfo.mTrelCounters.mTxBytes == 0);
+    TEST_ASSERT(trelInfo.mTrelCounters.mTxPackets == 0);
+    TEST_ASSERT(trelInfo.mTrelCounters.mTxFailure == 0);
```

Comment:
I think it'll be more comprehensive if you test the counters in a real use case. E.g. `test_trel_connectivity.py`. You can refer to `test_multi_thread_networks.py` about verifying the counters.

Reply:
I added some test assertions for counters in `test_trel_connectivity.py` in https://github.com/openthread/openthread/pull/9710 as part of the test for cli support of trel counters.

Reply:
I saw the python test case in the PR was using CLI. I'd suggest to call D-BUS in the test case so it will also cover the D-BUS implementation which is more similar to real use cases. 

Overall I'm fine with the testing in both PRs. Feel free to make the above change in future PRs if you'd like to.

Reply:
Sure, let's do it in the future pr.